### PR TITLE
Cinnamon Maven recipe

### PR DESCRIPTION
**Pull request by TimMoore**
Adapts the [Java + sbt](https://developer.lightbend.com/docs/cinnamon/current/getting-started/lagom_java.html) example for integrating Cinnamon into a Lagom service.

### CODE REVIEW

**Comment by TimMoore** [comment]
Don't delete this branch right after merging… wait for me to do it. I'm going to send a link to it to someone who asked for help with this.

**Comment by TimMoore** [comment]
CC: @bantonsson, @pvlugter, @henrikengstrom

**Comment by TimMoore** [comment]
You can browse this at https://github.com/lagom/lagom-recipes/tree/cinnamon-maven-example/lightbend-telemetry/lightbend-telemetry-java-mvn

**Comment by TimMoore** [comment]
Also, please create a merge commit and don't squash this when merging. The commit history is worth preserving, since it shows a step-by-step progression.

File: lightbend-telemetry/lightbend-telemetry-java-mvn/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+* Maven 3.3 or higher
+* Bintray credentials
+
+### Bintray credentials
+
+Follow [these instructions](https://www.lightbend.com/product/lightbend-reactive-platform/credentials) to set up your Bintray credentials for Maven.
+
+## Testing the recipe
+
+Lagom has a special development mode for rapid development, and does not fork the JVM when using the `lagom:runAll` or `lagom:run` commands in Maven. A forked JVM is necessary to gain metrics for actors and HTTP calls, since those are provided by the Cinnamon Java Agent. This example uses the [`exec-maven-plugin`](https://www.mojohaus.org/exec-maven-plugin/) to run the Lagom production server, enabling the complete set of metrics to be displayed.
```
Comment (ignasi35) [inline lightbend-telemetry/lightbend-telemetry-java-mvn/README.md]: This is a bit `meh`.

We're saying this recipe demonstrates using cinnamon in lagom but only for a single service and out of dev mode.

@pvlugter @bantonsson would it be possible to add the cinnamon agent on the JVM running `sbt` or `mvn` using `.jvmopts` or `MVN_OPTS`?

Comment (ignasi35) [reply lightbend-telemetry/lightbend-telemetry-java-mvn/README.md]: I'll merge. We can continue the discussion somewhere else.

Comment (pvlugter) [reply lightbend-telemetry/lightbend-telemetry-java-mvn/README.md]: Yes, should certainly be improved. And we've been here before with agents and Play dev run.

So the current plan is to have a classloader hook in the dev run, so Cinnamon agent can be added as a classloader transformation instead. This is how we used to have the Typesafe Console integration with Play dev run. This setting was removed. Issue to bring it back is here:

https://github.com/playframework/playframework/issues/8206

For Activator, which had both the Typesafe Console instrumentation (when it was open sourced) and integration with New Relic and AppDynamics agents, there was a forked dev run:

https://github.com/playframework/playframework/pull/3835

**Pull request merged**